server/repository/user/getOne.ts

```diff
@@ -6,29 +6,25 @@ const fields: Array<string> = ['lang', 'id', 'name', 'status', 'position', 'emai
 
 const selectFields = fields.map((f) => `u.${f}`).join(',')
 
-// TODO: pass props: Pick<User, 'email'>  | Pick<User, 'id'> | {emailGoogle:string}
-// TODO: rename method getOneOrNone - check if you can use getOne
-// TODO: insert in user auth provider google email during invitation
-// TODO: when passing emailGoogle, left join with users_auth_provider and where condition becomes props->>'email' = $1
-
-export const read = async (
-  props: { user: Pick<User, 'email'> } | { user: Pick<User, 'id'> },
+export const getOne = async (
+  props: { user: Pick<User, 'email'> | Pick<User, 'id'>; emailGoogle?: string },
```

Comment:
this should be Pick<User, 'email'> | Pick<User, 'id'> | {emailGoogle:string}

Reply:
the problem is that using Pick the descrutrurized prop user (const { user } = props) should be a User type object

Reply:
then do not pass user at all!
{id:number} | {email:string} | {emailGoogle:string}

Reply:
but this way is impossible to destructurize too. if props is defined this way is impossible to do props.id or { id } = props

Reply:
maybe this way { id?: number, email?: string, emailGoogle?: boolean } and check if one of them is defined at the beginning

Reply:
@sorja what do you think about it?

Reply:
ok

Reply:
https://github.com/openforis/fra-platform/pull/1201/commits/92212f47baf3399ff6b1636a14b93b41349250f6